doc/adr/2023-10-18 subdomain routing with cert per unit.md

```diff
@@ -0,0 +1,110 @@
+# Obtain a separate cert per every unit when using subdomain routing
+
+## Context and Problem Statement
+Until now, the TLS cert solution was designed with only path routing in mind.
+When using Traefik in the subdomain routing mode, all Ingress will use
+Traefik's default certificate, because charm code only obtains one cert (for
+the external hostname), and traefik obviously cannot match that cert with
+subdomains.
+
+https://github.com/canonical/traefik-k8s-operator/issues/244
+
+## Considered Options
+
+- One cert per app/unit (no wildcards)
+- Only one cert (as many wildcard SANs as necessary)
+- Only one cert (keep appending subdomains; no wildcards)
+- One cert per app (wildcard SANs only for ingress-per-unit)
+- One cert per app (keep appending IPU subdomains; no wildcards)
+
+## Decision Outcome
+
+Chosen option: "One cert per app/unit (no wildcards)", because we must support
+the use-case of strict subdomain routing (no wildcards).
+
+### Consequences
+
+* Good, because no wildcard certs (more secure).
+* Good, because not limited by SAN count per cert.
+* Bad, because may get rate limited by the CA.
```

Comment:
Let's encrypt rate limit is 50 per domain per week (www.domain.tld and sub.domain.tld count as the same domain in the limit)
Seems pretty reasonable, unless for testing purpose chances are limited that may need more than that amount

Reply:
One drawback I see is that for manual certificate signing (using for example the manual-tls-certificates-operator), the signing is more difficult than a wildcard

Reply:
The rate limit seems also reasonable to me. It could be worse if we used `Only one cert` and keep modifying it, because if we are not careful the rate limit could break the only cert in use.

Overall, I think it is something we can live with.